static/src/js/components/KeywordPicker/KeywordPicker.jsx

```diff
@@ -219,14 +220,15 @@ const KeywordPicker = ({
   const handleSubmit = () => {
     const addedKeywords = addKeywords(finalSelectedList.splice(1))
     if (addedKeywords) {
-      value.push(addedKeywords)
+      formData.push(addedKeywords)
+    } else {
+      return
     }
 
-    setValue(value)
     setFinalSelectedList([])
     setFinalSelectedValue('')
     setDisableButton(true)
-    onChange(value)
+    onChange(formData)
```

Comment:
Just realized something here that we missed when we solved this issue. I think we need to move the onChange(formData) inside the if statement and keep the setStates. With the current change the state is not going to reset. 


It would look like this: 
 ```
if (addedKeywords) {
       formData.push(addedKeywords)
       onChange(formData)
     }
     setFinalSelectedList([])
     setFinalSelectedValue('')
     setDisableButton(true)

```

Reply:
Yeah makes sense